src/server/game/Entities/Player/Player.cpp

```diff
@@ -14882,16 +14882,24 @@ void Player::RewardQuest(Quest const* quest, uint32 reward, Object* questGiver,
     uint32 quest_id = quest->GetQuestId();
 
     for (uint8 i = 0; i < QUEST_ITEM_OBJECTIVES_COUNT; ++i)
-        if (quest->RequiredItemId[i])
-            DestroyItemCount(quest->RequiredItemId[i], quest->RequiredItemCount[i], true);
-
+    {
+        if (ItemTemplate const* itemTemplate = sObjectMgr->GetItemTemplate(quest->RequiredItemId[i]))
+        {
+            if (quest->RequiredItemCount[i] > 0 && (itemTemplate->Bonding == BIND_QUEST_ITEM || itemTemplate->Bonding == BIND_QUEST_ITEM1))
+                DestroyItemCount(quest->RequiredItemId[i], 9999, true, true);
+        }
+        else 
```

Comment:
I am fucking blind! Also, I just checked. I know why my second test passed "with this code". No make install after the test compile, so I was still using a core with my old code and the else if which checked for quest->RequiredItemId[i] *sigh*